extensions/wikia/Search/WikiaSearchController.class.php

```diff
@@ -556,6 +557,22 @@ protected function handleArticleMatchTracking( Wikia\Search\Config $searchConfig
 	}
 
 	/**
+	 * Determine whether we should use "Go" search for exact title matches.
+	 *
+	 * This supports both the user preference and has backwards compatibility for
+	 * Monobook's "Go" button and "go" URL parameter on search.
+	 *
+	 * @return bool
+	 */
+	private function useGoSearch() {
+		$default = $this->getUser()->getGlobalPreference( 'enableGoSearch' ) ? '0' : 'Search';
+		// Support for Monobook's "Go" button and functionality
+		$fulltext = $this->getVal( 'fulltext', $this->getVal( 'go', $default ) );
+
+		return $fulltext === '0' || $fulltext === 'Go';
```

Comment:
is this the right priority?
1. fulltext
2. go
3. enableGoSearch

Shouldn't this be reverted? So user's preferences are the most important once?